service/src/main/kotlin/io/provenance/api/domain/usecase/provenance/account/models/GetSignerRequest.kt

```diff
@@ -1,9 +1,35 @@
 package io.provenance.api.domain.usecase.provenance.account.models
 
+import io.provenance.api.models.entity.Entity
 import io.provenance.api.models.account.AccountInfo
+import io.provenance.api.models.entity.KongConsumer
+import io.provenance.api.models.entity.MemberUUID
 import java.util.UUID
 
-data class GetSignerRequest(
-    val uuid: UUID,
+sealed interface GetSignerRequest {
+
+    companion object {
+        operator fun invoke(id: Entity, account: AccountInfo): GetSignerRequest = when (id) {
+            is KongConsumer -> GetSignerByAddressRequest(id.customId, account)
+            is MemberUUID -> GetSignerByUUIDRequest(id.value, account)
+        }
+    }
+
     val account: AccountInfo
-)
+
+    val entity: String
+        get() = when (this) {
+            is GetSignerByAddressRequest -> address
+            is GetSignerByUUIDRequest -> uuid.toString()
+        }
+}
+
+data class GetSignerByUUIDRequest(
+    val uuid: UUID,
+    override val account: AccountInfo
+) : GetSignerRequest
+
+data class GetSignerByAddressRequest(
+    val address: String,
+    override val account: AccountInfo
+) : GetSignerRequest
```

Comment:
Nit: why implement this in the interface rather than having the children implement it? e.g.
```suggestion
    val entity: String
}

data class GetSignerByUUIDRequest(
    val uuid: UUID,
    override val account: AccountInfo
) : GetSignerRequest {
    override val entity: String
        get() = uuid.toString()
}

data class GetSignerByAddressRequest(
    val address: String,
    override val account: AccountInfo
) : GetSignerRequest {
    override val entity: String
        get() = address
}
```

Reply:
Don't think I had a reason. I'll update to this suggestion